template.js

```diff
@@ -0,0 +1,86 @@
+const getRequestHeader = require('getRequestHeader');
+const sendHttpRequest = require('sendHttpRequest');
+const JSON = require('JSON');
+const encodeUriComponent = require('encodeUriComponent');
+const Math = require('Math');
+const logToConsole = require('logToConsole');
+const getContainerVersion = require('getContainerVersion');
+
+
+const apiUrl = "https://api.openweathermap.org/data/2.5/weather?";
+const units = data.units;
+const apiKey = data.apiKey;
+const isLoggingEnabled = determinateIsLoggingEnabled();
+const traceId = isLoggingEnabled ? getRequestHeader('trace-id') : undefined;
+function getCity() {
+    const cityFromHeaders = getRequestHeader('X-Geo-City');
+    if(cityFromHeaders === 'ZZ' || !cityFromHeaders) {
+        return null;
+    } else {
+        return cityFromHeaders;
+    }
+}
+
+const city = getCity();
+if(!city) return null;
+const url = apiUrl + "q=" + enc(city) + "&appid=" + enc(apiKey) + "&units=" + enc(units);
```

Comment:
look like it good Idea to provide country also as one city can be in multiple countries.